### ISSUE TITLE
Error with new API of PackageManager added in api level33 at robolectric tests.

### ISSUE THREAD

**Issue by masa-gclue** [robolectric/robolectric#7670]
### Description
The following API in PackageManager cause a "not supporting" error.
https://developer.android.com/reference/android/content/pm/PackageManager#getReceiverInfo(android.content.ComponentName,%20android.content.pm.PackageManager.ComponentInfoFlags)

https://developer.android.com/reference/android/content/pm/PackageManager#queryBroadcastReceivers(android.content.Intent,%20android.content.pm.PackageManager.ResolveInfoFlags)

https://developer.android.com/reference/android/content/pm/PackageManager#getServiceInfo(android.content.ComponentName,%20android.content.pm.PackageManager.ComponentInfoFlags)

```
sorry, not supporting public abstract android.content.pm.ActivityInfo android.content.pm.IPackageManager.getReceiverInfo(android.content.ComponentName,long,int) throws android.os.RemoteException yet!
java.lang.UnsupportedOperationException: sorry, not supporting public abstract android.content.pm.ActivityInfo android.content.pm.IPackageManager.getReceiverInfo(android.content.ComponentName,long,int) throws android.os.RemoteException yet!
	at org.robolectric.shadows.ShadowActivityThread$1.invoke(ShadowActivityThread.java:87)
	at com.sun.proxy.$Proxy35.getReceiverInfo(Unknown Source)
	at android.app.ApplicationPackageManager.getReceiverInfo(ApplicationPackageManager.java:581)
	at com.test.PackageManagerTest.getReceiverInfoWithApilevel33(PackageManagerTest.java:38)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:59)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:56)
	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
	at org.junit.runners.ParentRunner$3.evaluate(ParentRunner.java:306)
	at org.robolectric.RobolectricTestRunner$HelperTestRunner$1.evaluate(RobolectricTestRunner.java:580)
	at org.robolectric.internal.SandboxTestRunner$2.lambda$evaluate$2(SandboxTestRunner.java:287)
	at org.robolectric.internal.bytecode.Sandbox.lambda$runOnMainThread$0(Sandbox.java:99)
	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
	at java.base/java.lang.Thread.run(Thread.java:829)


sorry, not supporting public abstract android.content.pm.ParceledListSlice android.content.pm.IPackageManager.queryIntentReceivers(android.content.Intent,java.lang.String,long,int) throws android.os.RemoteException yet!
java.lang.UnsupportedOperationException: sorry, not supporting public abstract android.content.pm.ParceledListSlice android.content.pm.IPackageManager.queryIntentReceivers(android.content.Intent,java.lang.String,long,int) throws android.os.RemoteException yet!
	at org.robolectric.shadows.ShadowActivityThread$1.invoke(ShadowActivityThread.java:87)
	at com.sun.proxy.$Proxy35.queryIntentReceivers(Unknown Source)
	at android.app.ApplicationPackageManager.queryBroadcastReceiversAsUser(ApplicationPackageManager.java:1512)
	at android.app.ApplicationPackageManager.queryBroadcastReceivers(ApplicationPackageManager.java:1533)
	at com.test.PackageManagerTest.queryBroadcastReceiversWithApilevel33(PackageManagerTest.java:52)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:59)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:56)
	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
	at org.junit.runners.ParentRunner$3.evaluate(ParentRunner.java:306)
	at org.robolectric.RobolectricTestRunner$HelperTestRunner$1.evaluate(RobolectricTestRunner.java:580)
	at org.robolectric.internal.SandboxTestRunner$2.lambda$evaluate$2(SandboxTestRunner.java:287)
	at org.robolectric.internal.bytecode.Sandbox.lambda$runOnMainThread$0(Sandbox.java:99)
	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
	at java.base/java.lang.Thread.run(Thread.java:829)


sorry, not supporting public abstract android.content.pm.ServiceInfo android.content.pm.IPackageManager.getServiceInfo(android.content.ComponentName,long,int) throws android.os.RemoteException yet!
java.lang.UnsupportedOperationException: sorry, not supporting public abstract android.content.pm.ServiceInfo android.content.pm.IPackageManager.getServiceInfo(android.content.ComponentName,long,int) throws android.os.RemoteException yet!
	at org.robolectric.shadows.ShadowActivityThread$1.invoke(ShadowActivityThread.java:87)
	at com.sun.proxy.$Proxy35.getServiceInfo(Unknown Source)
	at android.app.ApplicationPackageManager.getServiceInfo(ApplicationPackageManager.java:604)
	at com.test.PackageManagerTest.getServiceInfoWithApilevel33(PackageManagerTest.java:66)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:59)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:56)
	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
	at org.junit.runners.ParentRunner$3.evaluate(ParentRunner.java:306)
	at org.robolectric.RobolectricTestRunner$HelperTestRunner$1.evaluate(RobolectricTestRunner.java:580)
	at org.robolectric.internal.SandboxTestRunner$2.lambda$evaluate$2(SandboxTestRunner.java:287)
	at org.robolectric.internal.bytecode.Sandbox.lambda$runOnMainThread$0(Sandbox.java:99)
	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
	at java.base/java.lang.Thread.run(Thread.java:829)

```

### Steps to Reproduce

```java
@RunWith(RobolectricTestRunner.class)
public class PackageManagerTest {
	@Test
	@Config(sdk = Build.VERSION_CODES.TIRAMISU)
	public void getReceiverInfoWithApilevel33() throws PackageManager.NameNotFoundException {
		Context context = ApplicationProvider.getApplicationContext();
		PackageManager packageManager = context.getPackageManager();
		ComponentName componentName = new ComponentName(context, MyReceiver.class);

		ActivityInfo result;
		// no error occurs.
		result = packageManager.getReceiverInfo(componentName, PackageManager.GET_META_DATA);
		assertThat(result, is(notNullValue()));

		// error here.
		result = packageManager.getReceiverInfo(componentName, PackageManager.ComponentInfoFlags.of(PackageManager.GET_META_DATA));
	}

	@Test
	@Config(sdk = Build.VERSION_CODES.TIRAMISU)
	public void queryBroadcastReceiversWithApilevel33() {
		Context context = ApplicationProvider.getApplicationContext();
		PackageManager packageManager = context.getPackageManager();
		Intent intent = new Intent("myapp.action.ACTION");

		List<ResolveInfo> result;
		// no error occurs.
		result = packageManager.queryBroadcastReceivers(intent, 0);
		assertThat(result, is(notNullValue()));

		// error here.
		result = packageManager.queryBroadcastReceivers(intent, PackageManager.ResolveInfoFlags.of(0));
	}

	@Test
	@Config(sdk = Build.VERSION_CODES.TIRAMISU)
	public void getServiceInfoWithApilevel33() throws PackageManager.NameNotFoundException {
		Context context = ApplicationProvider.getApplicationContext();
		PackageManager packageManager = context.getPackageManager();
		ComponentName componentName = new ComponentName(context, MyService.class);

		ServiceInfo result;
		// no error occurs.
		result = packageManager.getServiceInfo(componentName, PackageManager.GET_META_DATA);
		assertThat(result, is(notNullValue()));

		// error here.
		result = packageManager.getServiceInfo(componentName, PackageManager.ComponentInfoFlags.of(PackageManager.GET_META_DATA));
	}
}
```

### Robolectric & Android Version
- Robolectric 4.9
- Android SDK 33 (T)

### Link to a public git repo demonstrating the problem:
N/A

**Comment by utzcoz** [comment]
Hi @masa-gclue , thanks for reporting this problem. I checked code, and these methods are not supported by Robolectric now. Would you like to help to support it?

**Comment by MGaetan89** [comment]
I'll look into providing a PR for this issue

**Comment by MGaetan89** [comment]
The `queryBroadcastReceivers` method has been added in https://github.com/robolectric/robolectric/commit/9e923264af5d75067b38a3b2853bfacc0a9fde9b, and is available since Robolectric 4.11.
The other two are still missing, so I'll add them in an upcoming PR.
